pkgs/by-name/bl/bleep/package.nix

```
@@ -0,0 +1,91 @@
+{
+  stdenv,
+  stdenvNoCC,
+  fetchurl,
+  autoPatchelfHook,
+  installShellFiles,
+  makeWrapper,
+  lib,
+  zlib,
+}:
+let
+  platform =
+    {
+      x86_64-linux = "x86_64-pc-linux";
+      x86_64-darwin = "x86_64-apple-darwin";
+      aarch64-darwin = "arm64-apple-darwin";
+    }
+    ."${stdenv.system}";
+
+  hash =
+    {
+      x86_64-linux = "sha256-ttfmTMSK6u2zX87CUYjibr675Xzn4/byWub67e5Ir+I=";
+      x86_64-darwin = "sha256-bh1yQ7KZ+cwVHTLG+gpYDLPKG2VUcrz88Sqd2tRiKfM=";
+      aarch64-darwin = "sha256-IV2JPXibv80Odb+1Zkw1YUoutHsi5YaIjyNpAOIUlr4=";
+    }
+    ."${stdenv.system}";
+  version = "0.0.7";
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "bleep";
+  version = version;
+
+  src = fetchurl {
+    url = "https://github.com/oyvindberg/bleep/releases/download/v${version}/bleep-${platform}.tar.gz";
+    sha256 = hash;
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ] ++ lib.optional stdenv.isLinux autoPatchelfHook;
+
+  buildInputs = [
+    zlib
+    stdenv.cc.cc
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    tar -xf $src
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 bleep $out/bin/.bleep-wrapped
+    makeWrapper $out/bin/.bleep-wrapped $out/bin/bleep \
+      --argv0 "$out/bin/bleep"
+    runHook postInstall
+  '';
+
+  dontAutoPatchelf = true;
+
+  postFixup =
+    lib.optionalString stdenv.isLinux ''
+      autoPatchelf $out
+    ''
+    + ''
+      mkdir temp
+      cp $out/bin/.bleep-wrapped temp/bleep
+      PATH="./temp:$PATH"
+
+      installShellCompletion --cmd bleep \
+        --bash <(bleep install-tab-completions-bash --stdout) \
+        --zsh <(bleep install-tab-completions-zsh --stdout) \
+    '';
+
+  meta = {
+    homepage = "https://bleep.build/";
+    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
+    license = lib.licenses.mit;
+    description = "Bleeping fast scala build tool";
+    mainProgram = "bleep";
+    platforms = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
```

You should probably directly reference the ones in https://github.com/NixOS/nixpkgs/blob/master/lib/systems/doubles.nix

```suggestion
    platforms = with lib.platforms [
      x86_64-linux
      x86_64-darwin
      aarch64-darwin
    ];
```

---

I tried this, but it only gives me undefined variable. I'm not sure how I am supposed to refer to them?

---

You are right, disregard :sweat_smile: